storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -356,13 +358,14 @@ public static void sleep(long millis) {
         try {
             Time.sleep(millis);
         } catch(InterruptedException e) {
+            currentThread().interrupt();
```

Comment:
Here's a suboptimal example of a loop that will never exit as intended
`while(!currentThread.isInterrupted()) {
	execute();
}
...

void execute() {
	try {
		... // RE may be thrown here
		Utils.sleep(1000);
		... // and here
	} catch (RuntimeException e)
		log.error(e);
	}
}`